src/modules/job-info/job_state.c

```diff
@@ -1282,7 +1340,67 @@ static int journal_submit_event (struct job_state_ctx *jsctx,
     return job_transition_state (jsctx,
                                  job,
                                  FLUX_JOB_STATE_DEPEND,
-                                 timestamp);
+                                 timestamp,
+                                 0,
+                                 0);
+}
+
+static int priority_context_parse (flux_t *h,
+                                   struct job *job,
+                                   json_t *context)
+{
+    unsigned int priority;
+
+    if (!context
+        || json_unpack (context, "{ s:i }", "priority", (int *)&priority) < 0
+        || (priority < FLUX_JOB_PRIORITY_MIN
+            || priority > FLUX_JOB_PRIORITY_MAX)) {
```

Comment:
LGTM.com is calling these comparisons out because they can never be true.

I wonder if you should use `json_int_t priority;` and do away with the cast in `json_unpack()` and cast on assignment to `job->priority`?

Reply:
The cast is not needed anyway, since the varargs arguments are not type checked, right?

Since the priority is internally generated, it probably isn't necessary to bounds check here, especially when the valid range is the entire range of the data type

So my vote would be to drop the cast, and drop the check.

Reply:
sounds good.  As an aside, I completely missed that LGTM flagged something.  the check shows up as green/passing.  Is it based on an alert being a "error" vs "warning" kind of alert?

Reply:
Yeah, I've never paid much attention, it must be something like that -- but I can't remember if I've ever seen LGTM generate a "failed" status.

I noticed because of the comments LGTM adds to the PR.